tools/test-framework/src/chain/ext/wait_chain.rs

```diff
@@ -0,0 +1,51 @@
+use ibc_relayer_types::core::ics02_client::height::Height;
+
+use crate::prelude::*;
+
+/// Wait for a chain to get to a desired height, and timeout if
+/// the chain didn't get to that height in the desired amount of
+/// time.
+pub fn wait_for_chain_height<ChainA: ChainHandle, ChainB: ChainHandle>(
+    foreign_clients: &ForeignClientPair<ChainA, ChainB>,
+    target_reference_application_height: Height,
+    timeout_bound: Duration,
+) -> Result<(), Error> {
+    // Query the latest height of the chain
+    let mut reference_application_latest_height = foreign_clients
+        .client_a_to_b
+        .src_chain()
+        .query_latest_height()
+        .ok()
+        .unwrap();
+
+    let start = std::time::SystemTime::now();
+    while reference_application_latest_height != target_reference_application_height {
```

Comment:
Let's perhaps to this to avoid having the chain skip over the target height? It's unlikely to occur but just to be safe.

```suggestion
    while reference_application_latest_height < target_reference_application_height {
```